packages/runtime-vapor/__tests__/directives.spec.ts

```diff
@@ -0,0 +1,420 @@
+      expect(host.children[0]).toBe(el)
+
+      assertBindings(binding)
+    }) as DirectiveHook)
+
+    const unmounted = vi.fn(((el, binding) => {
+      expect(el.tagName).toBe('DIV')
+      // should have been removed
+      expect(el.parentNode).toBe(null)
+      expect(host.children.length).toBe(0)
+
+      assertBindings(binding)
+    }) as DirectiveHook)
+
+    const dir = {
+      beforeMount,
+      mounted,
+      beforeUpdate,
+      updated,
+      beforeUnmount,
+      unmounted,
+    }
+
+    let _instance: ComponentInternalInstance | null = null
+    let _node: Node | null = null
+    const Comp: Component = {
+      setup() {
+        _instance = currentInstance
+      },
+      render() {
+        _node = template('<div>')()
+        renderEffect(() => {
+          setText(_node!, count.value)
+        })
+        withDirectives(_node, [
+          [
+            dir,
+            // value
+            () => count.value,
+            // argument
+            'foo',
+            // modifiers
+            { ok: true },
+          ],
+        ])
+        return _node
+      },
+    }
+
+    const { host, render } = define(Comp)
+    const { instance } = render()
+
+    expect(beforeMount).toHaveBeenCalledTimes(1)
+    expect(mounted).toHaveBeenCalledTimes(1)
+
+    count.value++
+    await nextTick()
+    expect(beforeUpdate).toHaveBeenCalledTimes(1)
+    expect(updated).toHaveBeenCalledTimes(1)
+
+    unmountComponent(instance!)
+    expect(beforeUnmount).toHaveBeenCalledTimes(1)
+    expect(unmounted).toHaveBeenCalledTimes(1)
+  })
+
+  it('should work with a function directive', async () => {
+    const count = ref(0)
+
+    function assertBindings(binding: DirectiveBinding) {
+      expect(binding.value).toBe(count.value)
+      expect(binding.arg).toBe('foo')
+      expect(binding.instance).toBe(_instance)
+      expect(binding.modifiers && binding.modifiers.ok).toBe(true)
+    }
+
+    const fn = vi.fn(((el, binding) => {
+      expect(el.tagName).toBe('DIV')
+      expect(el.parentNode).toBe(host)
+
+      assertBindings(binding)
+    }) as DirectiveHook)
+
+    let _instance: ComponentInternalInstance | null = null
+    let _node: Node | null = null
+    const Comp = {
+      setup() {
+        _instance = currentInstance
+      },
+      render() {
+        _node = template('<div>')()
+        renderEffect(() => {
+          setText(_node!, count.value)
+        })
+        withDirectives(_node, [
+          [
+            fn,
+            // value
+            () => count.value,
+            // argument
+            'foo',
+            // modifiers
+            { ok: true },
+          ],
+        ])
+        return _node
+      },
+    }
+
+    const { host, render } = define(Comp)
+    render()
+
+    expect(fn).toHaveBeenCalledTimes(1)
+
+    count.value++
+    await nextTick()
+    expect(fn).toHaveBeenCalledTimes(2)
+  })
+
+  it('should work on component vnode', async () => {
```

Comment:
```suggestion
  it('should work on components', async () => {
```

Vapor does not have `vnode`, so the name of this test needs to be changed.